src/Data/Foreign/Generic/Enum.purs

```diff
@@ -0,0 +1,126 @@
+module Data.Foreign.Generic.EnumEncoding where
+
+import Prelude
+
+import Control.Alt ((<|>))
+import Data.Foreign (F, Foreign, ForeignError(..), fail, readString, toForeign)
+import Data.Foreign.Generic.Types (Options, SumEncoding(..))
+import Data.Generic.Rep (class Generic, Argument, Constructor(Constructor), NoArguments(NoArguments), Product, Rec, Sum(Inr, Inl), from, to)
+import Data.Symbol (class IsSymbol, SProxy(..), reflectSymbol)
+import Partial.Unsafe (unsafeCrashWith)
+
+-- | A generic function to be used with "Enums", or sum types with only no-argument constructors. This is used for decoding from strings to one of the constructors, combined with the `constructorTagTransform` property of `SumEncoding`.
+genericDecodeEnum
+  :: forall a rep
+   . Generic a rep
+  => GenericDecodeEnum rep
+  => Options
```

Comment:
If we're only using `constructorTagTransform` here, let's only require a record with that one field.